### PR TITLE
#90 twitter oauth

### PR DESCRIPTION
**Pull request by denys-medynskyi**
add #90

### CODE REVIEW

**Review by sveredyuk — APPROVED** [review]

File: app/services/omniauth/authentication_service.rb [hunk 1]
```diff
@@ -3,9 +3,12 @@ class AuthenticationService
     delegate :authenticated?, to: :policy
 
     def initialize(params)
+      @params       = params
       @uid          = params[:uid]
       @provider     = params[:provider]
-      @first_name, @last_name, @email = params.fetch(:info, {}).values_at(:first_name, :last_name, :email)
+      @first_name   = params_info[:first_name] || twitter_name&.first
+      @last_name    = params_info[:last_name] || twitter_name&.last
```
Comment (kushniryb) [inline app/services/omniauth/authentication_service.rb]: How about resolving `first_name, last_name & email` in a separate class based on provider?
It looks okay at the moment but it'll get really messy once we start adding more providers

Comment (denys-medynskyi) [reply app/services/omniauth/authentication_service.rb]: I also thought about, but wanted to make it work first. ok. I will do it. Thanks

**Comment by denys-medynskyi** [comment]
@sveredyuk I need credentials to Facebook app to test facebook login. Can you give me ?

File: app/services/omniauth/authentication_service.rb [hunk 2]
```diff
@@ -3,9 +3,12 @@ class AuthenticationService
     delegate :authenticated?, to: :policy
 
     def initialize(params)
+      @params       = params
       @uid          = params[:uid]
-      @provider     = params[:provider]
-      @first_name, @last_name, @email = params.fetch(:info, {}).values_at(:first_name, :last_name, :email)
+      @provider     = params[:provider]&.to_sym
+      @first_name   = params_resolver.first_name
```
Comment (denys-medynskyi) [inline app/services/omniauth/authentication_service.rb]: @kushniryb what do you think about this ?

**Pull request merged**